bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1139 @@
+                throw new ApiException("putResource() parsing error", e);
+            }
+        } catch (ExecutionException | TimeoutException e) {
+            throw new ApiException("putResource() error sending request", e);
+        } finally {
+            throttleDone();
+        }
+    }
+
+    /**
+     * Try to register the application key with the hub. Use the given application key if one is provided; otherwise the
+     * hub will create a new one. Note: this requires an HTTP 1.1 client call.
+     *
+     * @param oldApplicationKey existing application key if any i.e. may be empty.
+     * @return the existing or a newly created application key.
+     * @throws HttpUnauthorizedException if the registration failed.
+     * @throws ApiException if there was a communications error.
+     * @throws InterruptedException
+     */
+    public String registerApplicationKey(@Nullable String oldApplicationKey)
+            throws HttpUnauthorizedException, ApiException, InterruptedException {
+        logger.debug("registerApplicationKey()");
+        String json = jsonParser.toJson((Objects.isNull(oldApplicationKey) || oldApplicationKey.isEmpty())
+                ? new CreateUserRequest(APPLICATION_ID)
+                : new CreateUserRequest(oldApplicationKey, APPLICATION_ID));
+        Request httpRequest = httpClient.newRequest(registrationUrl).method(HttpMethod.POST)
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .content(new StringContentProvider(json), MediaType.APPLICATION_JSON);
+        ContentResponse contentResponse;
+        try {
+            logger.trace("POST {} HTTP/1.1 >> {}", registrationUrl, json);
+            contentResponse = httpRequest.send();
+        } catch (TimeoutException | ExecutionException e) {
+            throw new ApiException("HTTP processing error", e);
+        }
+        int httpStatus = contentResponse.getStatus();
+        json = contentResponse.getContentAsString().trim();
+        logger.trace("HTTP/1.1 {} {} << {}", httpStatus, contentResponse.getReason(), json);
+        if (httpStatus != HttpStatus.OK_200) {
+            throw new ApiException("HTTP bad response");
+        }
+        try {
+            List<SuccessResponse> entries = jsonParser.fromJson(json, SuccessResponse.GSON_TYPE);
+            if (Objects.nonNull(entries) && !entries.isEmpty()) {
+                SuccessResponse response = entries.get(0);
+                Map<String, Object> responseSuccess = response.success;
+                if (Objects.nonNull(responseSuccess)) {
+                    String newApplicationKey = (String) responseSuccess.get("username");
+                    if (Objects.nonNull(newApplicationKey)) {
+                        return newApplicationKey;
+                    }
+                }
+            }
+        } catch (JsonParseException e) {
+            logger.debug("registerApplicationKey() error parsing JSON response:{}", json);
```

Comment:
Do you think this would be useful also?
```suggestion
            logger.debug("registerApplicationKey() error parsing JSON response:{}", json, e);
```

Reply:
Done